### PR TITLE
clang-tidy

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/value.cpp [hunk 1]
```diff
@@ -1013,8 +1013,8 @@ std::ostream& TimeValue::write(std::ostream& os) const {
 
 int64_t TimeValue::toInt64(size_t /*n*/) const {
   // Returns number of seconds in the day in UTC.
-  int64_t result = (time_.hour - time_.tzHour) * 60 * 60;
-  result += (time_.minute - time_.tzMinute) * 60;
+  auto result = int64_t(time_.hour - time_.tzHour) * 60 * 60;
```
Comment (piponazo) [inline src/value.cpp:1016]: I do not see any issue in the previous code. All the members in the `Time` struct are `int`s and we have enough precision there to not have issues with casting.

Furthermore you are starting to use the C-Style casts here, and I have a strong preference for the usage of C++ casts. 

I also think that the change in `xmpsidecar.cpp` is not necessary.

File: src/basicio.cpp [hunk 2]
```diff
@@ -643,8 +643,8 @@ class EXIV2API BlockMap {
 
 void MemIo::Impl::reserve(size_t wcount) {
   const size_t need = wcount + idx_;
-  size_t blockSize = 32 * 1024;  // 32768
-  const size_t maxBlockSize = 4 * 1024 * 1024;
+  auto blockSize = size_t(32) * 1024;  // 32768
```
Comment (piponazo) [inline src/basicio.cpp:646]: Same thing in this commit. I do not think these clang-tidy changes are bringing anything possitive to the codebase. If something, it is making the code a bit more difficult to read.

**Review by piponazo — CHANGES_REQUESTED** [review]
Hi @neheb , this time I am not convinced about the changes you are proposing here. I normally accept your changes because I see a clear benefit for readability and improvements in the codebase. In this case, I do not see any benefit. I would only accept the 1st commit.

File: src/value.cpp (hunk 1 above)
Comment (neheb) [reply src/value.cpp:1016]: Right. The particular warning is about implicit widening.

**Comment by neheb** [comment]
removed.

**Review by piponazo — APPROVED** [review]
LGTM, thanks!

**Pull request merged**